test/cont_test.cpp

```diff
@@ -0,0 +1,102 @@
+#include <gtest/gtest.h>
+#include <fstream>
+#include <unordered_map>
+#include "../include/graph.h"
+#include "util/graph_gen.h"
+#include "util/mat_graph_verifier.h"
+
+void test_continuous(unsigned nodes, unsigned long updates_per_sample, unsigned long samples) {
+  srand(time(NULL));
+  Graph g(nodes);
+  size_t total_edges = static_cast<size_t>(nodes - 1) * nodes / 2;
+  std::vector<bool> adj(total_edges);
+  unsigned long num_failure = 0;
+  for (unsigned long i = 0; i < samples; i++) {
+    std::cout << "Starting updates" << std::endl;
+    for (unsigned long j = 0; j < updates_per_sample; j++) {
+      unsigned edgei = rand() % nodes;
+      unsigned edgej = rand() % (nodes - 1);
+      if (edgei > edgej) {
+        std::swap(edgei, edgej);
+      } else {
+        edgej++;
```

Comment:
Could there be some weird edge case here where we do an integer overflow and end up with a self loop?
I think so. It's amazingly unlikely but you could just add an else if just in case.

Reply:
I guess this isn't possible assuming nodes < unsigned max so never mind